OpenSim/Simulation/Test/testStatesTrajectory.cpp

```diff
@@ -641,6 +641,87 @@ void testIntegrityChecks() {
     // and Z's both pass the check. 
 }
 
+void tableAndTrajectoryMatch(const Model& model,
+                             const TimeSeriesTable& table,
+                             const StatesTrajectory& states,
+                             std::vector<std::string> columns = {}) {
+
+    const auto stateNames = model.getStateVariableNames();
+
+    int numColumns = -1;
+    if (columns.empty()) {
+        numColumns = stateNames.getSize();
+    } else {
+        numColumns = columns.size();
+    }
+    SimTK_TEST(table.getNumColumns() == numColumns);
+    SimTK_TEST(table.getNumRows() == states.getSize());
+
+    const auto& colNames = table.getDependentsMetaData()
+            .getValueArrayForKey("labels");
+
```

Comment:
This is ugly too. You can now use `table.getColumnLabels()` you suggested I add to DataTable.